tex/chapters/06-MachineLearning.tex

```diff
@@ -0,0 +1,240 @@
+Rows from the table containing \textbf{only} these strategies are then used to train the model.
+In Figure \ref{fig:ml_process}, the \textcolor{sol-red}{Red} and \textcolor{sol-blue}{Blue} strategies were selected, resulting in rows being selected that contained information relating exclusively to the \textcolor{sol-red}{Red} and \textcolor{sol-blue}{Blue} strategies.
+Finally, all remaining rows that were not involved in training the model are collected.
+The model is then scored against this unseen data.
+
+\begin{sidewaysfigure}[htbp!]
+    \centering
+    \subfloat[The Ashlock Tournament]{\includestandalone[width=0.15\textwidth]{../img/AshlockTournament}\label{fig:ashlock_tourn}}
+    \subfloat[Training the Model]{\includestandalone[width=0.85\textwidth]{../img/MachineLearningProcess}\label{fig:model_training}}
+    \caption{The process for training the Machine Learning Model}
+    \label{fig:ml_process}
+\end{sidewaysfigure}
+
+
+
+\section{Assessing the Model}\label{sec:assessing_model}
+
+The actual Machine Learning was done using the python library \mintinline{python}{sk-learn} \cite{scikit-learn}.
+A choice of two models were implemented, Logistic Regression \cite{Schmidt2016, Yu2011} and Support Vector Classification \cite{Smola2004, Vandewalle1999}.
+The process for training the models is outlined in Section \ref{sec:training_model} and it is mentioned that a sample must be taken from the strategies used in the initial Ashlock Tournament.
+Clearly the larger the sample size, the better the model should perform as it has more data to learn from.
+In order to decide whether to use Logistic Regression or SVC, both models were trained/tested against many samples of varying sizes.
+
+Figure \ref{fig:score_4_sample_size} shows how both models performed when trained on increasing sample sizes.
+50 different samples were taken for each sample size and the models were re-trained and re-scored for each sample.
+The violin plots in Figure \ref{fig:score_4_sample_size} show the distributions of those scores.
+
+\begin{figure}[htbp!]
+    \centering
+    \subfloat[Model performance against sample size for Linear Regression]{\includegraphics[width=\linewidth]{../img/ML/score_against_sample_size_LR.png}\label{fig:lr-samplesize}}\\
+    \subfloat[Model performance against sample size for SVC]{\includegraphics[width=\linewidth]{../img/ML/score_against_sample_size_SVC.png}\label{fig:svc-samplesize}}
+    \caption{Violin Plots for both models}
+    \label{fig:score_4_sample_size}
+\end{figure}
+
+It can be seen that the violins in Figure \ref{fig:svc-samplesize} reach a maximum score faster the those in Figure \ref{fig:lr-samplesize}.
+Those is Figure \ref{fig:svc-samplesize} also show far less variance, and so the decision was made to use SVC as the model in any future work.
+A sample size between 30 and 40 was deemed acceptable, due to a good balance between high score, low variance and computational cost.
```

Comment:
Immediate question (either answer it here or be ready to answer it): what is the computational cost